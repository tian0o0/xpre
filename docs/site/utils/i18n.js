// component mixin
import { get } from '../../../src/utils';
import { camelize } from '../../../src/utils/format/string';

export default {
  computed: {
    $t() {
      const { name } = this.$options;
      const prefix = name ? camelize(name) + '.' : '';
      const messages = this.$vantMessages[this.$vantLang];

      return (path, ...args) => {
        const message = get(messages, prefix + path) || get(messages, path);
        return typeof message === 'function' ? message(...args) : message;
      };
    }
  }
};
