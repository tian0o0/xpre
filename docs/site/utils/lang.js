import Locale from '../../../src/components/locale';
import zhCN from '../../../src/components/locale/lang/zh-CN';
import enUS from '../../../src/components/locale/lang/en-US';

const langMap = {
  'en-US': {
    title: 'Xpre - Exploring Endless Possibilities',
    messages: enUS
  },
  'zh-CN': {
    title: 'Xpre | 一个差不多的组件库',
    messages: zhCN
  }
};

let currentLang = '';

function getDefaultLang() {
  const langs = Object.keys(langMap);
  const { hash } = location;

  for (let i = 0; i < langs.length; i++) {
    if (hash.indexOf(langs[i]) !== -1) {
      return langs[i];
    }
  }

  const userLang = localStorage.getItem('VANT_LANGUAGE') || navigator.language || 'en-US';
  return userLang.indexOf('zh-') !== -1 ? 'zh-CN' : 'en-US';
}

export function setLang(lang) {
  if (currentLang === lang) {
    return;
  }

  currentLang = lang;
  if (window.localStorage) {
    localStorage.setItem('VANT_LANGUAGE', lang);
  }
  Locale.use(lang, langMap[lang].messages);
  document.title = langMap[lang].title;
}

setLang(getDefaultLang());
