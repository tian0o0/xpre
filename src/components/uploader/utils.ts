export function toArray<T>(item: T | T[]): T[] {
  if (Array.isArray(item)) {
    return item;
  }

  return [item];
}

export function readFile(file: File, resultType: string) {
  return new Promise(resolve => {
    const reader = new FileReader();

    reader.onload = event => {
      resolve((event.target as FileReader).result);
    };

    if (resultType === 'dataUrl') {
      reader.readAsDataURL(file);
    } else if (resultType === 'text') {
      reader.readAsText(file);
    }
  });
}

export function isOversize(files: File | File[], maxSize: number): boolean {
  return toArray(files).some(file => file.size > maxSize);
}

export type FileListItem = {
  url?: string;
  file?: File;
  content?: string; // dataUrl
};

const IMAGE_EXT = ['jpeg', 'jpg', 'gif', 'png', 'svg'];

export function isImageUrl(url: string): boolean {
  return IMAGE_EXT.some(ext => url.indexOf(`.${ext}`) !== -1);
}

export function isImageFile(item: FileListItem): boolean {
  if (item.file && item.file.type) {
    return item.file.type.indexOf('image') === 0;
  }

  if (item.url) {
    return isImageUrl(item.url);
  }

  if (item.content) {
    return item.content.indexOf('data:image') === 0;
  }

  return false;
}
