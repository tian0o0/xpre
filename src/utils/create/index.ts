import { createBEM, Bem } from './bem';
import { createComponent, Component } from './component';
import { createI18N, Translate } from './i18n';

type CreateNamespaceReturn = [
  Component,
  Bem,
  Translate
];

export function createNamespace(name: string): CreateNamespaceReturn {
  name = 'x-' + name;
  return [createComponent(name), createBEM(name), createI18N(name)];
}
