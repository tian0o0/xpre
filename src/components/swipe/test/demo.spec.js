import Demo from '../demo';
import demoTest from '../../../../test/demo-test';
import { mockGetBoundingClientRect } from '../../../../test/utils';

let restore;

demoTest(Demo, {
  hookBeforeTest: () => {
    restore = mockGetBoundingClientRect({
      width: 100,
      height: 100
    });
  },
  hookAfterTest: () => {
    restore();
  }
});
