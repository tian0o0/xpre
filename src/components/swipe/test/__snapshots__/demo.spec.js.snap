// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div>
  <div>
    <div class="x-swipe">
      <div class="x-swipe__track" style="width: 400px; transition-duration: 0ms; transform: translateX(0px);">
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">1</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">2</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">3</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">4</div>
      </div>
      <div class="x-swipe__indicators"><i class="x-swipe__indicator x-swipe__indicator--active" style="background-color: white;"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="x-swipe">
      <div class="x-swipe__track" style="width: 400px; transition-duration: 0ms; transform: translateX(0px);">
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);"><img data-src="https://img.yzcdn.cn/vant/apple-1.jpg" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);"><img data-src="https://img.yzcdn.cn/vant/apple-2.jpg" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);"><img data-src="https://img.yzcdn.cn/vant/apple-3.jpg" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);"><img data-src="https://img.yzcdn.cn/vant/apple-4.jpg" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      </div>
      <div class="x-swipe__indicators"><i class="x-swipe__indicator x-swipe__indicator--active"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="x-swipe">
      <div class="x-swipe__track" style="width: 400px; transition-duration: 0ms; transform: translateX(0px);">
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">1</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">2</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">3</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">4</div>
      </div>
      <div class="x-swipe__indicators"><i class="x-swipe__indicator x-swipe__indicator--active" style="background-color: white;"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="demo-swipe--vertical x-swipe" style="height: 200px;">
      <div class="x-swipe__track" style="height: 400px; transition-duration: 0ms; transform: translateY(0px);">
        <div class="x-swipe-item" style="width: 100px; height: 100px; transform: translateY(0px);">1</div>
        <div class="x-swipe-item" style="width: 100px; height: 100px; transform: translateY(0px);">2</div>
        <div class="x-swipe-item" style="width: 100px; height: 100px; transform: translateY(0px);">3</div>
        <div class="x-swipe-item" style="width: 100px; height: 100px; transform: translateY(0px);">4</div>
      </div>
      <div class="x-swipe__indicators x-swipe__indicators--vertical"><i class="x-swipe__indicator x-swipe__indicator--active" style="background-color: white;"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="x-swipe">
      <div class="x-swipe__track" style="width: 1200px; transition-duration: 0ms; transform: translateX(0px);">
        <div class="x-swipe-item" style="width: 300px; height: 100%; transform: translateX(0px);">1</div>
        <div class="x-swipe-item" style="width: 300px; height: 100%; transform: translateX(0px);">2</div>
        <div class="x-swipe-item" style="width: 300px; height: 100%; transform: translateX(0px);">3</div>
        <div class="x-swipe-item" style="width: 300px; height: 100%; transform: translateX(0px);">4</div>
      </div>
      <div class="x-swipe__indicators"><i class="x-swipe__indicator x-swipe__indicator--active" style="background-color: white;"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i><i class="x-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="x-swipe">
      <div class="x-swipe__track" style="width: 400px; transition-duration: 0ms; transform: translateX(0px);">
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">1</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">2</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">3</div>
        <div class="x-swipe-item" style="width: 100px; height: 100%; transform: translateX(0px);">4</div>
      </div>
      <div class="custom-indicator">1/4</div>
    </div>
  </div>
</div>
`;
